test/user.js

```diff
@@ -0,0 +1,161 @@
+var _Promise = require('bluebird');
+
+var chai = require('chai');
+var sinon = require('sinon');
+
+var errors = require('../api/v1/errors');
+var utils = require('../api/v1/utils');
+var UserService = require('../api/v1/services/UserService.js');
+var User = require('../api/v1/models/User.js');
+
+var assert = chai.assert;
+var expect = chai.expect;
+
+describe('UserService',function(){
+	describe('createUser', function () {
+		var _createUser;
+		var _findByEmail;
+
+		before(function (done) {
+			var testUser = User.forge({ id: 1, email: 'new@example.com' });
+			testUser.setPassword('password123').then(function () {
+				testUser.related('roles').add({ role: utils.roles.ATTENDEE });
+
+				_createUser = sinon.stub(User, 'create');
+				_findByEmail = sinon.stub(User, 'findByEmail');
+
+				_createUser.withArgs(sinon.match.string, sinon.match.string, sinon.match.string).returns(_Promise.resolve(testUser));
+				_findByEmail.withArgs('existing@example.com').returns(_Promise.resolve(testUser));
+				_findByEmail.withArgs(sinon.match.string).returns(_Promise.resolve(null));
+
+				done();
+			});
+		});
+		it('creates a new user', function (done) {
+			var user = UserService.createUser('new@example.com', 'password123', utils.roles.ATTENDEE);
+			expect(user).to.eventually.have.deep.property("attributes.id", 1);
+			expect(user).to.eventually.have.deep.property("attributes.email", 'new@example.com');
+
+			// TODO restructure this
```

Comment:
Do we still need to do this? I remember we talked about a better way of making it work.